controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -145,6 +145,7 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 		return ctrl.Result{}, nil
 	}
 	logger = logger.WithValues("cluster", cluster.Name)
+	r.managementCluster.SetClusterKeyIfEmpty(util.ObjectKey(cluster))
```

Comment:
Is `managementCluster` a field shared between all `KubeadmControlPlaneReconciler`, I don't think we should change a shared object this way, unless I'm missing something

Reply:
oh hmmm, that's a good point, gimme a few, you can probably stop reviewing at this point (because this will definitely have to change)

Reply:
added wip for now

Reply:
good call, thanks